File: leetcode/0881-boats-to-save-people/numRescueBoats.ts
function numRescueBoats(people: number[], limit: number): number {
  let boats = 0;
  people.sort((a, b) => a - b);

  for (
    let lightIndex = 0, heavyIndex = people.length - 1;
    lightIndex <= heavyIndex;
    heavyIndex--
  ) {
    const light = people[lightIndex];
    const heavy = people[heavyIndex];

    if (light + heavy <= limit) {
      lightIndex++;
    }

    boats++;
  }

  return boats;
}

export default numRescueBoats;

File: leetcode/0881-boats-to-save-people/numRescueBoats.go
package leetcode

import "sort"

func numRescueBoats(people []int, limit int) int {
	var boats = 0

	sort.Ints(people)

	for lightIndex, heavyIndex := 0, len(people)-1; lightIndex <= heavyIndex; heavyIndex-- {
		var light = people[lightIndex]
		var heavy = people[heavyIndex]

		if light+heavy <= limit {
			lightIndex++
		}

		boats++
	}

	return boats
}
